tests/config/test_config.py

```diff
@@ -111,6 +113,44 @@ def test_unknown_py_version(capsys: CaptureFixture) -> None:
     assert "the-newest has an invalid format, should be a version string." in output.err
 
 
+CSV_REGEX_COMMA_CASES = [
+    ("foo", ["foo"]),
+    ("foo,bar", ["foo", "bar"]),
+    ("foo, bar", ["foo", "bar"]),
+    ("foo, bar{1,3}", ["foo", "bar{1,3}"]),
+]
+
+
+@pytest.mark.parametrize("in_string,expected", CSV_REGEX_COMMA_CASES)
+def test_csv_regex_comma_in_quantifier(in_string, expected) -> None:
+    """Check that we correctly parse a comma-separated regex when there are one
+    or more commas within quantifier expressions.
+    """
+
+    def _template_run(in_string):
+        r = Run(
+            [str(EMPTY_MODULE), rf"--bad-names-rgx={in_string}"],
+            exit=False,
+        )
+        return r.linter.config.bad_names_rgxs
+
+    assert _template_run(in_string) == [re.compile(regex) for regex in expected]
+
+    # Catch trivially nonlinear performance
+    small_input_time = timeit.timeit(
+        "_template_run(in_string*100)",
+        globals=locals(),
+        number=10,
+    )
+    large_input_time = timeit.timeit(
+        "_template_run(in_string*1000)",
+        globals=locals(),
+        number=10,
+    )
+    fudge_factor = 3
+    assert large_input_time < small_input_time * 10 * fudge_factor
```

Comment:
We probably don't need to check for performance for a lot of parsing because it's always going to be parsed a dozen time per configuration read. What we could want to check is very long regex with a lot of opening and closing bracket (?). I think it's also overkill (especially if we deprecate now and remove in 3.0) but I would put that hypothetical test under a mark so it's executed only when we benchmark and it does not make all the tests longer.

```suggestion
```

Reply:
Sure. I just put it there because the most obvious implementation (to me) was very slow, so I found the test useful.